charts/spire/charts/spiffe-oidc-discovery-provider/templates/pre-delete-hook.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: v1
```

Comment:
Of the templates under spire-server/templates, the hook code is near 20% of the chart. 271 lines.

I've had reviewers complain to me about the maintenance cost of 3 line patches.

I've also had my own pr's asked to break out ingress objects via templates because its just too much duplication. which I have done.

I appreciate wanting to get things through. But there is a lot of inconsistency in the review of things. If we're going to have some rules for some, we should be consistent with their application. I've been on the short end of the stick a lot regarding this.

But, I'd also like to see smaller, faster pr's. So if we can commit to changing the process to be more open to duplication of stuff in order to get stuff merged faster, I'm game. Are you willing to give me that benefit as well in the future?

Reply:
Concistency in the implementation is more valueable then doing this implementation slightly different then the others.

All for improving it, but not in this PR as that goes way beyond the scope.

Reply:
Ok, can we then agree that duplication in a PR is OK so long as it is duplication for the sake of consistency with existing code? And in other cases (net new code), duplication is frowned upon?

Reply:
Ok. Followon pr it is.

Reply:
Taking your :+1: reaction as agreement (please let me know if it isn’t!). Don’t mean to belabor the point but just want to set some clear boundaries so I know how to best work with the project going forward

Reply:
Agree there is a lot of duplication with the hooks. Opened new issue to track de duplication effort: https://github.com/spiffe/helm-charts/issues/355

Reply:
> Taking your 👍 reaction as agreement (please let me know if it isn’t!). Don’t mean to belabor the point but just want to set some clear boundaries so I know how to best work with the project going forward

If the PR title can't capture the essence the PR is to big. This PR was just about adding the required hook to fix the uninstall bug. As PR titles are used in release notes it would be better to do a `Refactor helm hooks for less duplication` to improve on that.

Having these small targeted PRs without side effects/changes makes it also easier to review.

Only did the bare minimum refactor for the macro.

On the other hand I'm also ok with speccing out this piece of yaml for the RBAC stuff. It makes it way more transparant then abstracting everything in a macro which increases the cognitive complexity for any future contributor.